src/urllib3/connection.py

```diff
@@ -230,22 +309,32 @@ def request_chunked(self, method, url, body=None, headers=None):
 
         if body is not None:
             if isinstance(body, (str, bytes)):
-                body = (body,)
-            for chunk in body:
-                if not chunk:
-                    continue
-                if not isinstance(chunk, bytes):
-                    chunk = chunk.encode("utf8")
-                len_str = hex(len(chunk))[2:]
-                to_send = bytearray(len_str.encode())
-                to_send += b"\r\n"
-                to_send += chunk
-                to_send += b"\r\n"
-                self.send(to_send)
+                self.send_chunk(body)
+            else:
+                for chunk in body:
+                    self.send_chunk(chunk)
 
         # After the if clause, to always have a closed body
         self.send(b"0\r\n\r\n")
 
+    def send_chunk(self, chunk: Union[None, bytes, str]) -> None:
+        if not chunk:
+            return
+        if not isinstance(chunk, bytes):
+            chunk = chunk.encode("utf8")
+        len_str = hex(len(chunk))[2:]
+        to_send = bytearray(len_str.encode())
+        to_send += b"\r\n"
+        to_send += chunk
+        to_send += b"\r\n"
+        self.send(to_send)
+
+
+_PCTRTT = Tuple[Tuple[str, str], ...]
+_PCTRTTT = Tuple[_PCTRTT, ...]
+_PeerCertRetDictType = Dict[str, Union[str, _PCTRTTT, _PCTRTT]]
```

Comment:
We can maybe use `typing_extensions` and `TypedDict(..., total=False)` here to be more exact about the minimum types that come from `.getpeercert()`?

Reply:
Do you have something like:
```
class _PeerCertRetDictType(TypedDict, total=False):
    issuer: _PCTRTTT
    notAfter: str
    notBefore: str
    serialNumber: str
    subject: _PCTRTTT
    subjectAltName: _PCTRTT
    version: str
```
in mind?